### ISSUE TITLE
Segmentation fault reading block parameters on Mac

### ISSUE THREAD

**Issue by lxmota** [cmhamel/Exodus.jl#161]
SEGFAULT while trying to read block parameters. Succeeds when using read_block_parameters but segfaults when using read_set_parameters:
```
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.3 (2023-08-24)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Exodus

julia> exo = ExodusDatabase("cube.g", "r")
ExodusDatabase:
  File name                   = cube.g
  Mode                        = r
  Use cache arrays (advanced) = false

Initialization:
  Number of dim       = 3
  Number of nodes     = 27
  Number of elem      = 8
  Number of blocks    = 1
  Number of node sets = 6
  Number of side sets = 6

Block:
  cube

NodeSet:
  nsx+                            nsx-                            nsy+                            nsy-
  nsz+                            nsz-

SideSet:
  ssx+                            ssx-                            ssy+                            ssy-
  ssz+                            ssz-



julia> Exodus.read_set_parameters(exo, 1, Block)

[28097] signal (11.2): Segmentation fault: 11
in expression starting at REPL[3]:1
nc_utf8proc_iterate at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
nc_utf8proc_decompose_custom at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
nc_utf8proc_map_custom at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
nc_utf8proc_map at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
nc_utf8_normalize at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
NC_finddim at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
NC3_inq_dimid at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
nc_inq_dimid at /Users/amota/.julia/artifacts/9dfe83655d0b4372d36be73ef74acc1f79a12272/lib/libnetcdf.19.dylib (unknown line)
ex__get_dimension at /Users/amota/.julia/artifacts/a80dd849a9924a43293af9fde027eefdd9a66e39/lib/libexodus.2.0.dylib (unknown line)
ex_get_set_param at /Users/amota/.julia/artifacts/a80dd849a9924a43293af9fde027eefdd9a66e39/lib/libexodus.2.0.dylib (unknown line)
read_set_parameters at /Users/amota/.julia/packages/Exodus/nfL5g/src/Sets.jl:73
unknown function (ip: 0x11b80c16b)
ijl_apply_generic at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
do_call at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
eval_body at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
jl_interpret_toplevel_thunk at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
jl_toplevel_eval_flex at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
jl_toplevel_eval_flex at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
ijl_toplevel_eval_in at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
eval at ./boot.jl:370 [inlined]
eval_user_input at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-R17H3W25T9.0/build/default-honeycrisp-R17H3W25T9-0/julialang/julia-release-1-dot-9/usr/share/julia/stdlib/v1.9/REPL/src/REPL.jl:153
repl_backend_loop at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-R17H3W25T9.0/build/default-honeycrisp-R17H3W25T9-0/julialang/julia-release-1-dot-9/usr/share/julia/stdlib/v1.9/REPL/src/REPL.jl:249
#start_repl_backend#46 at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-R17H3W25T9.0/build/default-honeycrisp-R17H3W25T9-0/julialang/julia-release-1-dot-9/usr/share/julia/stdlib/v1.9/REPL/src/REPL.jl:234
start_repl_backend at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-R17H3W25T9.0/build/default-honeycrisp-R17H3W25T9-0/julialang/julia-release-1-dot-9/usr/share/julia/stdlib/v1.9/REPL/src/REPL.jl:231
ijl_apply_generic at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
#run_repl#59 at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-R17H3W25T9.0/build/default-honeycrisp-R17H3W25T9-0/julialang/julia-release-1-dot-9/usr/share/julia/stdlib/v1.9/REPL/src/REPL.jl:379
run_repl at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-honeycrisp-R17H3W25T9.0/build/default-honeycrisp-R17H3W25T9-0/julialang/julia-release-1-dot-9/usr/share/julia/stdlib/v1.9/REPL/src/REPL.jl:365
jfptr_run_repl_59992 at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
ijl_apply_generic at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
#1017 at ./client.jl:421
jfptr_YY.1017_37875 at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
ijl_apply_generic at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
jl_f__call_latest at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
#invokelatest#2 at ./essentials.jl:819 [inlined]
invokelatest at ./essentials.jl:816 [inlined]
run_main_repl at ./client.jl:405
exec_options at ./client.jl:322
_start at ./client.jl:522
jfptr__start_51456 at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/sys.dylib (unknown line)
ijl_apply_generic at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
true_main at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
jl_repl_entrypoint at /Applications/Julia-1.9.app/Contents/Resources/julia/lib/julia/libjulia-internal.1.9.dylib (unknown line)
Allocations: 642495 (Pool: 641904; Big: 591); GC: 1
Segmentation fault: 11
```

**Comment by cmhamel** [comment]
Could you attach the mesh to the issue @lxmota ?

**Comment by cmhamel** [comment]
@lxmota 

I did some digging in the exodus C side and I think the method I'm wrapping in ```read_set_parameters``` is not appropriate for blocks. I'm releasing a fix that will error if you try and use blocks in a way that doesn't segfault. 

You should use the ```read_block_parameters``` method.

**Comment by lxmota** [comment]
More info after today's chat:

```
(@v1.9) pkg> up Exodus#v0.10.1
    Updating registry at `~/.julia/registries/General.toml`
  No Changes to `~/.julia/environments/v1.9/Project.toml`
  No Changes to `~/.julia/environments/v1.9/Manifest.toml`

julia>
amota@s1094950ca:~/Repos/norma/examples/single/static-solid/cube (main=)
$ julia ../../../../src/norma.jl cube.yaml
Reading simulation file: cube.yaml
Initializing run at stop 0 with time = 0.00e+00
Establishing initial equilibrium
ERROR: LoadError: cannot resize array with shared data
Stacktrace:
  [1] _deleteend!
    @ ./array.jl:1023 [inlined]
  [2] resize!
    @ ./array.jl:1254 [inlined]
  [3] read_ids(exo::Exodus.ExodusDatabase{Int32, Int32, Int32, Float64}, #unused#::Type{Exodus.Block})
    @ Exodus ~/.julia/packages/Exodus/nfL5g/src/Sets.jl:11
  [4] read_sets(exo::Exodus.ExodusDatabase{Int32, Int32, Int32, Float64}, type::Type{Exodus.Block})
    @ Exodus ~/.julia/packages/Exodus/nfL5g/src/Sets.jl:181
  [5] evaluate(#unused#::Main.Norma.QuasiStatic, model::Main.Norma.SolidMechanics)
    @ Main.Norma ~/Repos/norma/src/model.jl:178
  [6] evaluate(integrator::Main.Norma.QuasiStatic, solver::Main.Norma.HessianMinimizer, model::Main.Norma.SolidMechanics)
    @ Main.Norma ~/Repos/norma/src/solver.jl:186
  [7] solve(integrator::Main.Norma.QuasiStatic, solver::Main.Norma.HessianMinimizer, model::Main.Norma.SolidMechanics)
    @ Main.Norma ~/Repos/norma/src/solver.jl:276
  [8] initialize
    @ ~/Repos/norma/src/time_integrator.jl:96 [inlined]
  [9] initialize(sim::Main.Norma.SingleDomainSimulation)
    @ Main.Norma ~/Repos/norma/src/evolve.jl:100
 [10] evolve(sim::Main.Norma.SingleDomainSimulation)
    @ Main.Norma ~/Repos/norma/src/evolve.jl:5
 [11] run
    @ ~/Repos/norma/src/norma.jl:8 [inlined]
 [12] top-level scope
    @ ~/Repos/norma/src/norma.jl:19
in expression starting at /Users/amota/Repos/norma/src/norma.jl:1
amota@s1094950ca:~/Repos/norma/examples/single/static-solid/cube (main=)
```

**Comment by cmhamel** [comment]
@lxmota I patched some things up in #165 which was the likely culprit for the issue you encountered. I'm putting out a new release (Exodus v0.11.0) momentarily so you can try this out when you get the chance. I haven't had a chance to try the specific norma example.

**Comment by lxmota** [comment]
@cmhamel Sorry for the late reply. I was out of the US and I returned last night. I updated to the latest version and now norma runs again. Thanks!
